openedx/features/survey_report/views.py

```diff
@@ -0,0 +1,31 @@
+"""
+Views to manage the Survey Reports.
+"""
+
+
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.generic import View
+from common.djangoapps.util.views import ensure_valid_course_key
+from .tasks import generate_survey_report
+from .api import generate_report
+
+
+class SurveyReportView(View):
+    """
+    View for Survey Reports.
+    """
+    @method_decorator(login_required)
+    @method_decorator(ensure_csrf_cookie)
+    @method_decorator(ensure_valid_course_key)
+    def post(self, request):
+        """
+        Generate a new survey report using the generate_report method in api.py
+        Arguments:
+            request: HTTP request
+        """
+        survey_report_id = generate_report(defaults=True)
+        generate_survey_report.delay(survey_report_id)
```

Comment:
I think I have an example of this. It involves an exception instead of a simple slow middleware, but I think it holds:

```python

survey_report_id = generate_report(defaults=True)

    # SQL: with transaction;
    # SQL: insert `value` in generated_reports;

generate_survey_report.delay(survey_report_id) # -> new thread in a different server
return redirect("admin:survey_report_surveyreport_changelist")

    middleware 1
    middleware 2 -> exception
    middleware 3

    ## somewhere at the end of the middlewares
        if not exception:
            # SQL: commit;
        else:
            # SQL: rollback;
```

In the async server, since the connection is different we have no context of the transaction and the ID might not even exist.

Reply:
@ormsbee I've already moved all to the generate_report method, what do you think? the we will have the same problem with these changes or are enough?

Reply:
Yeah, I think it's fine now, thank you. Please just squash your commits and add the context from your PR message to your commit message. Thank you!